debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -1086,6 +1086,37 @@ public void readSchema(Tables tables, String databaseCatalog, String schemaNameP
         return pkColumnNames;
     }
 
+    protected List<String> readTableUniqueIndices(DatabaseMetaData metadata, TableId id) throws SQLException {
```

Comment:
Ok, so it seems this would be prone to race conditions in case of changes to the definition of a table's PK and/or unique key constraints?

Reply:
@gunnarmorling Unfrotunately yes, but this is for cases when the information cannot be obtained any other way

Reply:
Ok; but is there _any_ case for SQL Server where the unique constraints can be obtained without the OOB read?

Reply:
@gunnarmorling There is https://docs.microsoft.com/en-us/sql/relational-databases/system-stored-procedures/sys-sp-cdc-get-ddl-history-transact-sql?view=sql-server-2017

The problem is that we still need operator cooperation when changing data structure by adding new capture instance. IIRC the table captured only subset of DDLs so at the end we would need to apply parser and still do the OOB read which was not considered worth the complications. Morevore `CREATE INDEX` at all would not be captured too